File: backend/src/main/java/com/wealthmanagementsystem/dto/response/LoginResponse.java
package com.wealthmanagementsystem.dto.response;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Response DTO for user login.
 * 
 * Contains authentication token and user information.
 * 
 * @author Wealth Management System
 * @version 1.0
 * @since Phase 4.8 Batch 3
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class LoginResponse {
    
    private String token;
    
    private String tokenType = "Bearer";
    
    private Long userId;
    
    private String email;
}

File: frontend/src/types/auth/LoginResponse.ts
/**
 * Response payload returned after successful login.
 *
 * @since Phase 6.4
 */
export interface LoginResponse {
  /**
   * JWT access token returned after successful authentication.
   * This token should be stored and included in subsequent API requests.
   */
  token: string

  /**
   * Authentication scheme.
   * Example: "Bearer"
   */
  tokenType: string

  /**
   * Authenticated user's unique identifier.
   */
  userId: number

  /**
   * Authenticated user's email address.
   */
  email: string
}
